uefi/src/proto/device_path/mod.rs

```diff
@@ -332,6 +356,75 @@ impl DevicePath {
             stop_condition: StopCondition::EndEntireNode,
         }
     }
+
+    #[cfg(feature = "alloc")]
+    pub fn to_text_by_node(
+        &self,
+        bs: &BootServices,
+    ) -> crate::Result<Vec<(&DevicePathNode, CString16)>> {
+        use uefi::Identify;
+        let &handle = bs
+            .locate_handle_buffer(SearchType::ByProtocol(&DevicePathToText::GUID))
+            .unwrap()
+            .first()
+            .unwrap();
+
+        let to_text_protocol = unsafe {
+            bs.open_protocol::<DevicePathToText>(
+                OpenProtocolParams {
+                    handle,
+                    agent: bs.image_handle(),
+                    controller: None,
+                },
+                OpenProtocolAttributes::GetProtocol,
+            )
+        }
+        .unwrap();
```

Comment:
Why don't you propagate the error here? I know at least (older) apple firmware doesn't guarantee this protocol to be around.

Reply:
I omitted the error propagation for fast progress to quickly just get a few thoughts about the API. 

I will add proper error handling, if we decide that this is useful and the right thing to do